src/vhost_user/master.rs

```diff
@@ -452,6 +458,43 @@ impl VhostUserMaster for Master {
         Ok(())
     }
 
+    fn get_inflight_fd(&mut self) -> Result<(VhostUserInflight, RawFd)> {
+        let mut node = self.node();
+        if node.acked_protocol_features & VhostUserProtocolFeatures::INFLIGHT_SHMFD.bits() == 0 {
+            return error_code(VhostUserError::InvalidOperation);
+        }
+
+        let hdr = node.send_request_header(MasterReq::GET_INFLIGHT_FD, None)?;
```

Comment:
According to the C implementation of libvhost-user, the backend also needs the `VhostUserInflight` to create a memfd and mmap it with proper size.  When I perform testing, with QEMU's vhost-user-blk daemon, this absence of msg payload will cause the daemon to panic (see https://github.com/qemu/qemu/blob/7fe7fae8b48e3f9c647fd685e5155ebc8e6fb84d/subprojects/libvhost-user/libvhost-user.c#L1662).

Therefore, maybe we should create a `VhostUserInflight` and send it along with the header. (Reference from QEMU: https://github.com/qemu/qemu/blob/7fe7fae8b48e3f9c647fd685e5155ebc8e6fb84d/hw/virtio/vhost-user.c#L2305)

Reply:
Yes let me fix that :)